rust/ql/lib/codeql/rust/elements/internal/CallImpl.qll

```diff
@@ -0,0 +1,128 @@
+private import rust
+private import codeql.rust.internal.PathResolution
+private import codeql.rust.internal.TypeInference as TypeInference
+private import codeql.rust.elements.internal.ExprImpl::Impl as ExprImpl
+private import codeql.rust.elements.Operation
+
+module Impl {
+  /**
+   * An expression that calls a function.
+   *
+   * This class abstract over the different ways in which a function can be called in Rust.
+   */
+  abstract class Call extends ExprImpl::Expr {
```

Comment:
I wonder if this class should instead be added via `annotations.py` and suitable `replace_bases`?

Reply:
It seems to me that it would make sense to try an merge `Call` and `CallExprBase` into one class. But that will require adapting a lot of places, so I think it makes sense to defer that to a follow up PR.